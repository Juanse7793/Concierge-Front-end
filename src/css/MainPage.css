/* Main Page */
.title {
  font-size: 2.5rem;
}

.subtitle {
  font-size: 0.9rem;
  color: var(--theme-dark-grey);
}

/* Carausel / Transitions */
.events-list {
  --diameter: 250px;
  --growth: 4rem;

  align-items: center;
  justify-content: space-between;
  width: 100%;
}

.transit {
  position: relative;
  justify-content: space-around;
}

/* Event */
.event-card {
  align-items: center;
  justify-content: center;
  color: black;
  padding: 1rem;
  margin: 1rem;
  min-height: 420px;
}

.circle {
  position: relative;
  height: var(--diameter);
  width: var(--diameter);
  border-radius: 50%;
  overflow: hidden;
  border: var(--border);
  transition: all 1s ease-in-out;
}

.event-card:hover .circle {
  height: calc(var(--diameter) + var(--growth));
  width: calc(var(--diameter) + var(--growth));
  margin: -1rem;
  border-radius: 1rem;
  box-shadow: 0 0 0 8px white inset;
}

.circle > img {
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  object-position: center;
  transition: all 1s ease-in-out;
}

.event-card-details {
  margin: 0;
  font-size: 0;
  opacity: 0;
  transition: all 1s ease-in-out;
}

.event-card:hover .event-card-details {
  font-size: 1rem;
  opacity: 0.3;
}

@media (max-width: 768px) {
  .events-list {
    flex-direction: column;
  }

  .events-list .pill {
    --long-end: 1rem;
  }

  .events-list .pill:disabled {
    display: none;
  }

  .events-list .prev {
    align-self: flex-start;
  }

  .events-list .next {
    align-self: flex-end;
  }

  .event-card:hover {
    pointer-events: none;
  }

  .event-card-details {
    font-size: 1rem;
    opacity: 0.3;
  }
}
