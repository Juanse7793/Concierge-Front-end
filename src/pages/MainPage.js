import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import EventCard from '../components/EventCard';
import SemiBtn from '../components/SemiBtn';

const MainPage = () => {
  const events = useSelector((state) => state.events);
  const [slice, setSlice] = useState(0);
  const step = 3;
  const sliceEvents = events.slice(slice, slice + step);

  const prevSlice = () => {
    if (slice <= step) setSlice(0);
    else setSlice(slice - step);
  };
  const nextSlice = () => {
    if (slice >= events.length - step) setSlice(events.length - step);
    else setSlice(slice + step);
  };

  return (
    <div>
      <div className="title-box">
        <h1 className="title">Exiting New Events!</h1>
        <h2 className="subtitle">Please select an event</h2>
      </div>
      <div id="event-list">
        <SemiBtn arrow="prev" disabled={slice < step} func={prevSlice} />
        {sliceEvents.map((event) => (
          <EventCard
            key={event.id}
            name={event.name}
            image={event.image}
            location={event.location}
            price={event.price}
          />
        ))}
        <SemiBtn arrow="next" disabled={slice > events.length - step} func={nextSlice} />
      </div>
    </div>
  );
};

export default MainPage;
