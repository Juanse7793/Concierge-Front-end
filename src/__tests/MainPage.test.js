import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import MainPage from '../pages/MainPage';

const mockStore = configureMockStore();
const store = mockStore({
  events: {
    events: [
      {
        id: 1,
        name: 'foo',
        location: 'loc',
        start_date: '2022-01-01',
        end_date: '2023-01-01',
        price: 123,
        image_urls: ['images/foo.jpg'],
      },
      {
        id: 2,
        name: 'bar',
        location: 'ation',
        start_date: '2022-01-01',
        end_date: '2023-01-01',
        price: 123,
        image_urls: ['images/bar.jpg'],
      },
      {
        id: 3,
        name: 'foobar',
        location: 'location',
        start_date: '2022-01-01',
        end_date: '2023-01-01',
        price: 123,
        image_urls: ['images/foo.jpg', 'images/bar.jpg'],
      },
    ],
    adding: false,
  },
});

jest.mock('../redux/reducers/events', () => ({
  fetchEvents: () => ({ type: 'MOCK' }),
}));

it('matches snapshot', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <MainPage />
      </Router>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
