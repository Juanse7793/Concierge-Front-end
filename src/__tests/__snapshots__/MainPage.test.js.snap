// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="column"
>
  <div
    className="title-box"
  >
    <h1
      className="title green-glow"
    >
      EXCITING NEW EVENTS!
    </h1>
    <h2
      className="subtitle"
    >
      Please select an event to begin:
    </h2>
  </div>
  <div
    className="events-list row"
  >
    <button
      className="pill semi prev green"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      ◁
    </button>
    <div
      className="events-list row transit"
    >
      <a
        className="event-card column"
        href="/events/1"
        onClick={[Function]}
      >
        <div
          className="circle"
        >
          <img
            alt=""
            className="eventImage"
            src="images/foo.jpg"
          />
        </div>
        <h3
          className="event-card-title"
        >
          foo
        </h3>
        <p
          className="event-card-details"
        >
          loc
        </p>
        <p
          className="event-card-details"
        >
          $123
        </p>
      </a>
      <a
        className="event-card column"
        href="/events/2"
        onClick={[Function]}
      >
        <div
          className="circle"
        >
          <img
            alt=""
            className="eventImage"
            src="images/bar.jpg"
          />
        </div>
        <h3
          className="event-card-title"
        >
          bar
        </h3>
        <p
          className="event-card-details"
        >
          ation
        </p>
        <p
          className="event-card-details"
        >
          $123
        </p>
      </a>
    </div>
    <button
      className="pill semi next green"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      ▷
    </button>
  </div>
</section>
`;
